hero_python_port/tests/test_parameters.py

```
@@ -1,401 +1,259 @@
+        b=lambda p_ctx, c: p_ctx['a'] * 2,
+        c_fixed=1,
+        d=lambda p_ctx, c: p_ctx['b'] + p_ctx['c_fixed'] + c
     )
+    assert params.names == ['a', 'b', 'c_fixed', 'd']
+    results_cycle0 = params.evaluate_all_deterministic(cycle=0)
+    assert results_cycle0 == {'a': 5, 'b': 10, 'c_fixed': 1, 'd': 11}
+    results_cycle3 = params.evaluate_all_deterministic(cycle=3)
+    assert results_cycle3 == {'a': 5, 'b': 10, 'c_fixed': 1, 'd': 14}
+
+def test_evaluate_all_deterministic_with_probabilistic_dist():
+    norm_dist = MockNormal(mean=50, std=5)
```

Ruff (0.11.9)

Undefined name `MockNormal`

(F821)